pyproject.toml

```diff
@@ -29,6 +29,7 @@ strict_equality = true
 module = ["trieste.*", "tests.*"]
 disallow_any_generics = true
 disallow_untyped_defs = true
+disallow_subclassing_any = true
```

Comment:
what does this do?

Reply:
See https://mypy.readthedocs.io/en/stable/command_line.html#disallow-dynamic-typing:

> This flag reports an error whenever a class subclasses a value of type Any. This may occur when the base class is imported from a module that doesn’t exist (when using --ignore-missing-imports) or is ignored due to --follow-imports=skip or a # type: ignore comment on the import statement.
> 
> Since the module is silenced, the imported class is given a type of Any. By default mypy will assume that the subclass correctly inherited the base class even though that may not actually be the case. This flag makes mypy raise an error instead.

Reply:
so bcs tf.Module was Any mypy would report it as an error?

Reply:
That's right (and I checked that it did)